src/Controller/ResetController.php

```diff
@@ -116,16 +120,22 @@ public function resetPassword(
         UriSigner $uriSigner
     ): Response {
         if ($uriSigner->check(sprintf('%s://%s%s', $request->getScheme(), $request->getHttpHost(), $request->getRequestUri())) === false) {
-            $this->session->getFlashBag()->add('danger', 'Not possible to reset password, please request a reset again.');
+            $defaultResponse = new RedirectResponse($this->router->generate('connectholland_user_reset'));
+            $passwordResetFailedEvent = new PasswordResetFailedEvent($defaultResponse, 'danger', __FUNCTION__);
+            /* @scrutinizer ignore-call */
+            $this->eventDispatcher->dispatch(UserBundleEvents::PASSWORD_RESET_FAILED, $passwordResetFailedEvent);
 
-            return new RedirectResponse($this->router->generate('connectholland_user_reset'));
+            return $passwordResetFailedEvent->getResponse();
         }
 
         $user = $this->registry->getRepository(UserInterface::class)->findOneBy(['passwordRequestToken' => $token, 'email' => $email]);
         if ($user instanceof UserInterface === false) {
-            $this->session->getFlashBag()->add('danger', 'Not possible to reset password, please request a reset again.');
+            $defaultResponse = new RedirectResponse($this->router->generate('connectholland_user_reset'));
+            $passwordResetFailedEvent = new PasswordResetFailedEvent($defaultResponse, 'danger', __FUNCTION__);
```

Comment:
I prefer to use a (class) constant instead of `__FUNCTION__`. 
Change this function name will result in failure of an app, while renaming the controller function should normally not be a problem in a controller as it is found by the router.